pokemongo_bot/cell_workers/recycle_items.py

```diff
@@ -1,65 +1,162 @@
+    You can use either item's name or id. For the full list of items see ../../data/items.json
+
+    It's highly recommended to put this task before the move_to_fort task in the config file so you'll most likely be able to loot.
+
+    Example config :
+    {
+      "type": "RecycleItems",
+      "config": {
+        "item_filter": {
+          "Pokeball": {"keep": 20},
+          "Greatball": {"keep": 50},
+          "Ultraball": {"keep": 100},
+          "Potion": {"keep": 0},
+          "Super Potion": {"keep": 0},
+          "Hyper Potion": {"keep": 20},
+          "Max Potion": {"keep": 50},
+          "Revive": {"keep": 0},
+          "Max Revive": {"keep": 20},
+          "Razz Berry": {"keep": 20}
+        }
+      }
+    }
+    """
+
     def initialize(self):
-        self.item_filter = self.config.get('item_filter', {})
+        self.items_filter = self.config.get('item_filter', {})
         self._validate_item_filter()
 
     def _validate_item_filter(self):
+        """
+        Validate user's item filter config
+        :return: Nothing.
+        :rtype: None
+        :raise: ConfigException: When an item doesn't exist in ../../data/items.json
+        """
         item_list = json.load(open(os.path.join('data', 'items.json')))
-        for config_item_name, bag_count in self.item_filter.iteritems():
+        for config_item_name, bag_count in self.items_filter.iteritems():
             if config_item_name not in item_list.viewvalues():
                 if config_item_name not in item_list:
                     raise ConfigException("item {} does not exist, spelling mistake? (check for valid item names in data/items.json)".format(config_item_name))
 
     def work(self):
-        self.bot.latest_inventory = None
-        item_count_dict = self.bot.item_inventory_count('all')
-
-        for item_id, bag_count in item_count_dict.iteritems():
-            item_name = self.bot.item_list[str(item_id)]
-            id_filter = self.item_filter.get(item_name, 0)
-            if id_filter is not 0:
-                id_filter_keep = id_filter.get('keep', 20)
+        """
+        Discard items if necessary.
+        :return: Always returns WorkerResult.SUCCESS.
+        :rtype: WorkerResult
+        """
+        if not self.bot.has_space_for_loot():
+            # Updating user's inventory
+            self.bot.latest_inventory = None
+            # Getting every item in user's inventory
+            item_counts_in_bag_dict = self.bot.item_inventory_count('all')
+
+            # For each user's item in inventory recycle the item if needed
+            for item_id, item_count_in_bag in item_counts_in_bag_dict.iteritems():
+                item = RecycleItems._Item(item_id, self.items_filter, self)
+
+                if item.should_be_recycled():
+                    item.request_recycle()
+                    item.emit_recycle_result()
+
+        return WorkerResult.SUCCESS
+
+    class _Item:
```

Comment:
When you inherit from `object` you have new-style class. 

New style objects have a different object model to classic objects, and some things won't work properly with old style objects (those that doesn't inherit from `object`), for instance, `super()`, `@property` and `descriptors`.